rllib/evaluation/rollout_worker.py

```diff
@@ -1581,6 +1565,50 @@ def _build_policy_map(
             logger.info(f"Built policy map: {self.policy_map}")
             logger.info(f"Built preprocessor map: {self.preprocessors}")
 
+    def _get_make_sub_env_fn(self, env_creator, env_context, validate_env, env_wrapper, seed):
+
+        def _make_sub_env(vector_index):
+            # Used to created additional environments during environment
+            # vectorization.
+
+            # Create the env context (config dict + meta-data) for
+            # this particular sub-env within the vectorized one.
+            env_ctx = env_context.copy_with_overrides(
+                vector_index=vector_index)
+            # Create the sub-env.
+            env = env_creator(env_ctx)
+            # Validate first.
+            _validate_env(env, env_context=env_ctx)
+            # Custom validation function given by user.
+            if validate_env is not None:
+                validate_env(env, env_ctx)
+            # Use our wrapper, defined above.
+            env = env_wrapper(env)
+
+            # Make sure a deterministic random seed is set on
+            # all the sub-environments if specified.
+            _update_env_seed_if_necessary(env, seed, env_context.worker_index,
+                                          vector_index)
+            return env
+
+        if not env_context.remote:
+
+            def make_sub_env(vector_index):
+                sub_env = _make_sub_env(vector_index)
+                self.callbacks.on_sub_environment_created(
+                    worker=self,
+                    sub_environment=sub_env,
+                    env_context=env_context.copy_with_overrides(
+                        worker_index=env_context.worker_index,
+                        vector_index=vector_index,
+                        remote=False),
+                )
+                return sub_env
+        else:
+            make_sub_env = _make_sub_env
+
```

Comment:
sorry a quick coding style suggestion, can we name this function better, move it to the top level as the other version, so that it's clear these 2 functions are are the same level, and we are just picking one of them here?
now the names only differ by a leading _, and whole thing reads a bit confusing.
```
def _make_sub_env_local(...):
  ...

def _make_sub_env_remote(...):
  ...

if env_context.remote:
  return _make_sub_env_remote
else:
  return _make_sub_env_local
```
thanks.

Reply:
Oh, sorry, yeah this doesn't look easy to read :) Will fix.

Reply:
done